File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        i = 0
        j = len(height) - 1
        max_area = 0
        while i < j:
            h = min(height[i], height[j])
            width = j - i
            total = h * width
            max_area = max(max_area, total)
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return max_area

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        int max = 0;
        while (i < j) {
            int h = Math.min(height[i], height[j]);
            int width = j - i;
            int total = h * width;
            max = Math.max(max, total);
            if (height[i] < height[j]) {
                i++;
            } 
            else {
                j--;
            }
        }
        return max;
    }
}